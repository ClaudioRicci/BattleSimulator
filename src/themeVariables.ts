export const colors = {
  black: "#000",
  white: "#FFF",
  orange: "#f9b600",
  grey: "#F1F1F1",
  primary: "#00008D",
  secondary: "#F900F1",
  ternary: "#FB2020"
};

export const fonts = {
  header: "Verdana, Arial, Helvetica, sans-serif",
  paragraph: "Arial, Helvetica, sans-serif"
};

export const align = {
  left: "left",
  right: "right",
  center: "center"
};

export const deviceSizes = {
  mobile: "768px",
  desktop: "1024px"
};
