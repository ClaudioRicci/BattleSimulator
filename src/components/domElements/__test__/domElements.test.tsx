import React from "react";
import ReactDOM from "react-dom";
import { BoardSurround, PlayerSurround, ActionButton } from "../index";
import { colors, fonts, align, deviceSizes } from "../../../themeVariables";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

describe("Dom Element Tests", () => {
  describe("BoardSurround Tests", () => {
    it("BoardSurround renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<BoardSurround />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("BoardSurround to be 60rem wide by default", () => {
      const tree = renderer.create(<BoardSurround />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("width", "60rem");
    });
  });
  describe("ActionButton Tests", () => {
    it("ActionButton renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<ActionButton />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("ActionButton should have pointer to denote it can be pressed", () => {
      const tree = renderer.create(<ActionButton />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("cursor", "pointer");
    });

    test("ActionButton should turn orange (#f9b600) when hovered", () => {
      const tree = renderer.create(<ActionButton />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("color", `${colors.orange}`, {
        modifier: ":hover"
      });
    });

    test("ActionButton should be smaller for Mobile (Device Width 768px or smaller)", () => {
      const tree = renderer.create(<ActionButton />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule("width", "6.4rem", {
        media: `(max-width: ${deviceSizes.mobile})`
      });
    });
  });

  describe("Date Tests", () => {
    it("Date renders correctly", () => {
      const div = document.createElement("div");
      ReactDOM.render(<PlayerSurround />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    test("Date is secondary colour (#5D5D5D - Dark Grey)", () => {
      const tree = renderer.create(<PlayerSurround />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(tree).toHaveStyleRule(`color, ${colors.secondary}`);
    });
  });
});
