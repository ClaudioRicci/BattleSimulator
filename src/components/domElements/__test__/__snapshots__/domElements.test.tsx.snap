// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dom Element Tests ActionButton Tests ActionButton should be smaller for Mobile (Device Width 768px or smaller) 1`] = `
.c0 {
  background: #FFF;
  border: 0.2rem #00008D solid;
  color: #00008D;
  cursor: pointer;
  font: 900 0.8rem/1rem Verdana,Arial,Helvetica,sans-serif;
  margin: 1rem auto;
  outline: none;
  padding: 0.5rem;
}

.c0:hover {
  border: 0.2rem #f9b600 solid;
  color: #f9b600;
  opacity: 0.85;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

@media (max-width:768px) {
  .c0 {
    width: 6.4rem;
    left: 13.6rem;
    position: absolute;
    top: 4.7rem;
  }
}

<button
  className="c0"
/>
`;

exports[`Dom Element Tests ActionButton Tests ActionButton should have pointer to denote it can be pressed 1`] = `
.c0 {
  background: #FFF;
  border: 0.2rem #00008D solid;
  color: #00008D;
  cursor: pointer;
  font: 900 0.8rem/1rem Verdana,Arial,Helvetica,sans-serif;
  margin: 1rem auto;
  outline: none;
  padding: 0.5rem;
}

.c0:hover {
  border: 0.2rem #f9b600 solid;
  color: #f9b600;
  opacity: 0.85;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

@media (max-width:768px) {
  .c0 {
    width: 6.4rem;
    left: 13.6rem;
    position: absolute;
    top: 4.7rem;
  }
}

<button
  className="c0"
/>
`;

exports[`Dom Element Tests ActionButton Tests ActionButton should turn orange (#f9b600) when hovered 1`] = `
.c0 {
  background: #FFF;
  border: 0.2rem #00008D solid;
  color: #00008D;
  cursor: pointer;
  font: 900 0.8rem/1rem Verdana,Arial,Helvetica,sans-serif;
  margin: 1rem auto;
  outline: none;
  padding: 0.5rem;
}

.c0:hover {
  border: 0.2rem #f9b600 solid;
  color: #f9b600;
  opacity: 0.85;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

@media (max-width:768px) {
  .c0 {
    width: 6.4rem;
    left: 13.6rem;
    position: absolute;
    top: 4.7rem;
  }
}

<button
  className="c0"
/>
`;

exports[`Dom Element Tests BoardSurround Tests BoardSurround to be 60rem wide by default 1`] = `
.c0 {
  text-align: center;
  width: 60rem;
  height: 46rem;
  padding: 1.6rem 1.6rem 0.5rem;
  margin: 0 auto;
  background: #F1F1F1;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    width: 90%;
    height: 46rem;
  }
}

<div
  className="c0"
/>
`;

exports[`Dom Element Tests Date Tests Date is secondary colour (#5D5D5D - Dark Grey) 1`] = `
.c0 {
  width: 20rem;
  padding: 1.6rem 1.6rem 0.5rem;
  border: 0.25rem #00008D solid;
  background: #FFF;
  border-radius: 1rem;
}

<div
  className="c0"
/>
`;
